src/client/builder.rs

```diff
@@ -252,12 +252,18 @@ where
 
     let mut out = match parts.path_and_query {
         Some(p) => match p.query() {
-            Some(x) => format!("{}?{}", p.path(), x),
+            Some(query) => format!("{}?{}", p.path(), query),
             None => format!("{}?", p.path()),
         },
         None => "/?".to_string(),
     };
-    let mut serializer = form_urlencoded::Serializer::new(&mut out);
+    let mut serializer = if out.ends_with('?') {
+        let start_position = out.len();
+        form_urlencoded::Serializer::for_suffix(&mut out, start_position)
+    } else {
+        form_urlencoded::Serializer::new(&mut out)
+    };
```

Comment:
This code path had the same kind of bug, modified the relevant test to catch it.